Source/Scene/ModelExperimental/TextureManager.js

```diff
@@ -0,0 +1,170 @@
+import defined from "../../Core/defined.js";
+import destroyObject from "../../Core/destroyObject.js";
+import Texture from "../../Renderer/Texture.js";
+
+/**
+ * An object to manage loading textures
+ *
+ * @alias TextureManager
+ * @constructor
+ *
+ * @private
+ * @experimental This feature is using part of the 3D Tiles spec that is not final and is subject to change without Cesium's standard deprecation policy.
+ */
+export default function TextureManager() {
+  this._defaultTexture = undefined;
+  this._textures = {};
+  this._loadedImages = [];
+
+  // Keep track of the last time update() was called to avoid
+  // calling update() twice.
+  this._lastUpdatedFrame = -1;
+}
+
+/**
+ * Get one of the loaded textures
+ * @param {String} textureId The unique ID of the texture loaded by {@link TextureManager#loadTexture2D}
+ * @return {Texture} The texture or <code>undefined</code> if no texture exists
+ */
+TextureManager.prototype.getTexture = function (textureId) {
+  return this._textures[textureId];
+};
+
+function fetchTexture2D(textureManager, textureId, textureUniform) {
+  textureUniform.resource
+    .fetchImage()
+    .then(function (image) {
+      textureManager._loadedImages.push({
+        id: textureId,
+        image: image,
+        textureUniform: textureUniform,
+      });
+    })
+    .otherwise(function () {
+      var texture = textureManager._textures[textureId];
+      if (defined(texture) && texture !== textureManager._defaultTexture) {
+        texture.destroy();
+      }
+
+      textureManager._textures[textureId] = textureManager._defaultTexture;
+    });
+}
+
+/**
+ * Load a texture 2D asynchronously. Note that {@link TextureManager#update}
+ * must be called in the render loop to finish processing the textures.
+ *
+ * @param {String} textureId A unique ID to identify this texture.
+ * @param {TextureUniform} textureUniform A description of the texture
+ *
+ * @private
+ */
+TextureManager.prototype.loadTexture2D = function (textureId, textureUniform) {
+  if (defined(textureUniform.typedArray)) {
+    this._loadedImages.push({
+      id: textureId,
+      textureUniform: textureUniform,
+    });
+  } else {
+    fetchTexture2D(this, textureId, textureUniform);
+  }
+};
+
+function createTexture(textureManager, loadedImage, context) {
+  var id = loadedImage.id;
+  var textureUniform = loadedImage.textureUniform;
+
+  var typedArray = textureUniform.typedArray;
+  var sampler = textureUniform.sampler;
+
+  var texture;
+  if (defined(typedArray)) {
+    texture = new Texture({
+      context: context,
+      source: {
+        arrayBufferView: typedArray,
+        width: textureUniform.width,
+        height: textureUniform.height,
+      },
+      sampler: sampler,
+    });
```

Comment:
I think this should pass in `flipY: false`, assuming the bottom-left of the texture is the origin. The checkboard example in the custom shader sandcastle makes it a little hard to verify correctness.

And in case I missed it, the documentation should state the pixel order for typed array input, e.g. row major from bottom row to top row.

Reply:
ah I was wondering about this. Is this only for the typed array version of creating a Texture, or both?

Reply:
Just for the typed array version. PNG and JPEG are top-left origin so they would use the default `flipY: true`

Reply:
We talked about this, we want to make this flipY change and also add one more sandcastle example with a gradient to check the flip-y

Reply:
Just made this change and added a sandcastle demo. It's a little hard to tell from the model which way is up, but it does appear right-side-up in Spector:

![image](https://user-images.githubusercontent.com/8422414/131386061-1a20df40-55e7-412a-834d-1ba1e6386406.png)